src/tasks/predict_modality/workflows/process_datasets/main.nf

```diff
@@ -77,12 +78,40 @@ workflow run_wf {
       ]
     )
 
+    // extract the dataset metadata
+    | extract_metadata.run(
+      key: "extract_metadata_mod1",
+      fromState: [input: "output_train_mod1"],
+      toState: { id, output, state ->
+        state + [
+          modality_mod1: readYaml(output.output).uns.modality
+        ]
+      }
+    )
+
+    // extract the dataset metadata
+    | extract_metadata.run(
+      key: "extract_metadata_mod2",
+      fromState: [input: "output_train_mod2"],
+      toState: { id, output, state ->
+        state + [
+          modality_mod2: readYaml(output.output).uns.modality
+        ]
+      }
+    )
+
+    | map { id, state ->
+      def new_id = id + "_" + state.modality_mod1 + "2" + state.modality_mod2
```

Comment:
I think this new_id also needs to be passed to the process_data component so that it can be updated in the uns, no?

In case of doubt, let's discuss